adapter/denier/denier.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package denier
+
+// NOTE: This adapter will eventually be auto-generated so that it automatically supports all CHECK and QUOTA
+//       templates known to Mixer. For now, it's manually curated.
+
+import (
+	"context"
+	"time"
+
+	"github.com/golang/protobuf/proto"
+	rpc "github.com/googleapis/googleapis/google/rpc"
+
+	"istio.io/mixer/adapter/denier/config"
+	"istio.io/mixer/pkg/adapter"
+	"istio.io/mixer/template/checknothing"
+	"istio.io/mixer/template/listentry"
+	"istio.io/mixer/template/quota"
+)
+
+type (
+	builder struct{}
+
+	handler struct {
+		status rpc.Status
+	}
+)
+
+var _ checknothing.HandlerBuilder = builder{}
+var _ checknothing.Handler = handler{}
+var _ listentry.HandlerBuilder = builder{}
+var _ listentry.Handler = handler{}
+var _ quota.HandlerBuilder = builder{}
+var _ quota.Handler = handler{}
+
+///////////////// Configuration Methods ///////////////
+
+func (builder) Build(cfg proto.Message, env adapter.Env) (adapter.Handler, error) {
+	return handler{status: cfg.(*config.Params).Status}, nil
+}
+
+func (builder) ConfigureCheckNothingHandler(map[string]*checknothing.Type) error {
+	return nil
+}
+
+func (builder) ConfigureListEntryHandler(map[string]*listentry.Type) error {
+	return nil
+}
+
+func (builder) ConfigureQuotaHandler(map[string]*quota.Type) error {
+	return nil
+}
+
+////////////////// Runtime Methods //////////////////////////
+
+func (h handler) HandleCheckNothing(context.Context, *checknothing.Instance) (adapter.CheckResult, error) {
+	return adapter.CheckResult{
+		Status:        h.status,
+		ValidDuration: 1000 * time.Second,
+		ValidUseCount: 1000,
+	}, nil
+}
+
+func (h handler) HandleListEntry(context.Context, *listentry.Instance) (adapter.CheckResult, error) {
+	return adapter.CheckResult{
+		Status:        h.status,
+		ValidDuration: 1000 * time.Second,
+		ValidUseCount: 1000,
```

Comment:
No, we don't have unlimited at this point. The noop2 adapter sets the expiration to 10000000 or something. Here, I went for 1000 in order for updates in the returned error via config to be reflected in a running system eventually.